### PR TITLE
Set buffer localvars for channel and server

### LINKED ISSUE #188: Localvars for server and channel
Lots of scripts and logger settings and relay clients look at the buffer local variables server and channel.
Another useful localvar is short_name.

I think slack should set those on each buffer it creates.

### PR DESCRIPTION
**Pull request by trygveaa**
These are the same as what the irc plugin sets, and can be used in the
logger masks among other things.

This fixes #188.

### CODE REVIEW

**Comment by tfheen** [comment]
Looks fine to me. @rawdigits, ok to merge?

**Comment by trygveaa** [comment]
Hm, with this, the server var is set to `<team>.slack.com`. If my suggestion in https://github.com/rawdigits/wee-slack/issues/144#issuecomment-202512332 is done, it would be set to `slack.<team>`.

Maybe it should be set to only the team name instead? This would make it more like to the irc plugin, where it is only set to the server name (e.g. "EFNet").

**Comment by tfheen** [comment]
Just the team name would be fine, it'll be namespaced by the plugin name anyway, won't it? (For logging.)

**Comment by trygveaa** [comment]
Yeah, it makes more sense for logging as well. That way, you can choose if you want slack as part of the file name or not. I changed it.

File: wee_slack.py [hunk 1]
```diff
@@ -419,6 +421,8 @@ def create_buffer(self):
                 w.buffer_set(self.channel_buffer, "localvar_set_type", 'private')
             else:
                 w.buffer_set(self.channel_buffer, "localvar_set_type", 'channel')
+            w.buffer_set(self.channel_buffer, "localvar_set_channel", self.name)
+            w.buffer_set(self.channel_buffer, "localvar_set_server", self.server.team)
```
Comment (trygveaa) [inline wee_slack.py]: Should this take the server alias into account?

Comment (tfheen) [inline wee_slack.py]: ]] Trygve Aaberge 

> In wee_slack.py:
> 
> > @@ -419,6 +421,8 @@ def create_buffer(self):
> >                  w.buffer_set(self.channel_buffer, "localvar_set_type", 'private')
> >              else:
> >                  w.buffer_set(self.channel_buffer, "localvar_set_type", 'channel')
> > +            w.buffer_set(self.channel_buffer, "localvar_set_channel", self.name)
> > +            w.buffer_set(self.channel_buffer, "localvar_set_server", self.server.team)
> 
> Should this take the server alias into account?

Yes, I think so makes sense.

-- 
Tollef Fog Heen
UNIX is user friendly, it's just picky about who its friends are

Comment (trygveaa) [inline wee_slack.py]: Fixed.

**Pull request merged**